File: photomanager-api/src/main/java/at/hb/photomanager/dto/PmExifDto.java
package at.hb.photomanager.dto;

import lombok.Getter;
import lombok.Setter;
import org.joda.time.DateTime;

/**
 * @author blascheh
 */
@Getter
@Setter
public class PmExifDto {
    private Long height; // in pixels original
    private Long width; // in pixels original
    private Long widthTn; // in pixels thumbnail
    private Long heightTn; // in pixels thumbnail
    private Long xRes; // in dots
    private Long yRes; // in dots
    private Long orientation; // 0 unkown
    private DateTime date; // Aufnahmedatum
    private String filename;
    private Long filesize; // in Byte
    private DateTime modifiedDate; // geändert am
}

File: photomanager-angular-ui/src/app/model/PmExifDto.ts
import {PmBaseDto} from "./PmBaseDto";

export interface PmExifDto extends PmBaseDto{
    height: number;
    width: number;
    widthTn: number;
    heightTn: number;
    xRes: number;
    yRes: number;
    orientation: number;
    date: Date
    filename: string;
    filesize: number;
    modifiedDate: Date;
}


export function sortPicturesByAufnahmedatum(c1: PmExifDto, c2: PmExifDto) {
    return c1.date.getMilliseconds() - c2.date.getMilliseconds();
}
